corehq/apps/users/management/commands/gdpr_scrub_user_from_forms.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from django.core.management.base import BaseCommand
+from corehq.apps.users.models import CommCareUser
+from corehq.form_processor.interfaces.dbaccessors import FormAccessors
+from io import StringIO
+from lxml import etree
+import sys
+import six
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Scrubs the username from all forms associated with the given user"
+
+    def add_arguments(self, parser):
+        parser.add_argument('username')
+        parser.add_argument('domain')
+
+    def handle(self, username, domain, **options):
+        this_form_accessor = FormAccessors(domain=domain)
+        user = CommCareUser.get_by_username(username)
+        if not user:
+            logger.info("User {} not found.".format(username))
+            sys.exit(1)
+        user_id = user._id
+        form_ids = this_form_accessor.get_form_ids_for_user(user_id)
+        new_username = "Redacted User (GDPR)"
+        input_response = six.moves.input(
+            "Update {} form(s) for user {} in domain {}? (y/n): ".format(len(form_ids), username, domain))
+        if input_response == "y":
+            for form_data in this_form_accessor.iter_forms(form_ids):
+                form_attachment_xml_new = self.update_form_data(form_data, new_username)
+                this_form_accessor.modify_attachment_xml_and_metadata(form_data, form_attachment_xml_new)
+            logging.info("Updated {} form(s) for user {} in domain {}".format(len(form_ids), username, domain))
+        elif input_response == "n":
+            logging.info("No forms updated, exiting.")
+        else:
+            logging.info("Command not recognized. Exiting.")
+
+    @staticmethod
+    def update_form_data(form_data, new_username):
+        form_attachment_xml = form_data.get_attachment("form.xml")
+        xml_elem = etree.parse(StringIO(six.text_type(form_attachment_xml)))
+        id_elem = xml_elem.find("{http://openrosa.org/jr/xforms}meta").find(
+            "{http://openrosa.org/jr/xforms}username")
+        id_elem.text = new_username
```

Comment:
I still think string search and replace would be faster and more robust here. It is possible that a username could be saved to any question in a form.

Reply:
Weighing in here because I think an unstructured search/replace has some significant disadvantages. Suppose a project is scrubbing username `test@test.com` and there's another user in the domain named `my_test@test.com`—unless crafted very carefully (and email addresses are notoriously hard to parse) a search/replace may match things that should not be replaced. Also, unstructured search/replace could cause other parts of the metadata (or anything stored in couch or SQL or elastic that has been parsed from the XML) to become out of sync with the XML.

Reply:
Agreed ^, besides I checked with @krsdimagi who confirmed that instances of the username not specifically in the `username` tag are not in our scope